README.md

```diff
@@ -1,30 +1,112 @@
-# terraform-module-template
+# terraform-datadog-users
 
-[![Release](https://img.shields.io/github/release/masterpointio/terraform-module-template.svg)](https://github.com/masterpointio/terraform-module-template/releases/latest)
+[![Release](https://img.shields.io/github/release/masterpointio/terraform-datadog-users.svg)](https://github.com/masterpointio/terraform-datadog-users/releases/latest)
 
-This repository serves as a template for creating Terraform modules, providing a standardized structure and essential files for efficient module development. It's designed to ensure consistency and best practices across Terraform projects.
+This Terraform module is designed to create and manage Datadog user accounts, with a specific focus on security and sensitive data handling.
+
+To ensure the secure handling of DataDog sensitive data for datadog/datadog provider, the module is designed to work with the [SOPS (Secrets OPerationS) provider](https://github.com/mozilla/sops). SOPS is a tool for encrypting and decrypting files containing sensitive data, often used in conjunction with Terraform to manage secrets.
+
+The module utilizes [terraform-secrets-helper](https://github.com/masterpointio/terraform-secrets-helper/tree/main) module - a helper that provides a standard way of managing secrets from different sources, incuding SOPS files.
 
 ## Usage
 
-TODO
+```hcl
+module "datadog_users" {
+  source  = "masterpointio/datadog/users"
+  version = "X.X.X"
+
+  users = [
+    {
+      access_roles = {
+        "datadog" = true,
+        "aws"     = true,
+      },
+      email    = "jane.smith@example.com",
+      name     = "Jane Smith",
+      role     = "Editor",
+      username = "janesmith"
+    }
+  ]
+
+  secret_mapping = [
+    {
+      name = "datadog_api_key"
+      file = "example.yaml"
+      type = "sops"
+    },
+    {
+      name = "datadog_app_key"
+      file = "example.yaml"
+      type = "sops"
+    }
+  ]
+}
+```
+
+Check out [examples/complete](examples/complete) for the full example.
+
+❗ We recommend to use AWS KMS, GCP KMS, Azure Key Vault for SOPS files encryption. Don't use the secrets from the example in your real configuration!
+
+Here are some basic SOPS operations that help you to work with the example:
+
+- SOPS provider configuration for the example:
+
+```sh
+cd ./example/complete
+export SOPS_AGE_KEY_FILE=key.txt
+```
+
+- Encryption
+
+```sh
+sops --encrypt --age age1uafwjn52f8qvdeyqgn5epens4nwpqaqld8ln47xs8an003r3gudqxpzt7m example.raw.yaml > example.yaml
```

Comment:
This is purely a side note, but one thing I would suggest is that we never include the `*.raw.yaml` sops file creation idea. I have a feeling that is going to get people to check-in raw secrets. Instead, we should always push people towards the simple `sops <filename>` pattern where they will then never end up writing plain secrets to disk. Does that make sense?

Reply:
Yeah, totally. Updated to the `sops <filename>` pattern 👍